core/src/display_object/movie_clip.rs

```diff
@@ -1128,8 +1128,46 @@ impl<'gc> MovieClip<'gc> {
                         child.set_place_frame(context.gc_context, self.current_frame());
                     }
 
+                    // Apply PlaceObject parameters.
+                    child.apply_place_object(context, place_object);
+                    if let Some(name) = &place_object.name {
+                        let encoding = swf::SwfStr::encoding_for_version(self.swf_version());
+                        let name = name.to_str_lossy(encoding);
+                        child.set_name(
+                            context.gc_context,
+                            AvmString::new_utf8(context.gc_context, name),
+                        );
+                    }
+                    if let Some(clip_depth) = place_object.clip_depth {
+                        child.set_clip_depth(context.gc_context, clip_depth.into());
+                    }
+                    // Clip events only apply to movie clips.
+                    if let (Some(clip_actions), Some(clip)) =
```

Comment:
I think `clip` can be simply replaced with `self`, as it's already known to be a MovieClip.